lib/arboreal/instance_methods.rb

```diff
@@ -42,7 +42,8 @@ def root?
 
     # return the root of the tree
     def root
-      root? ? self : ancestors.first
+      return self if root?
+      (root_relation_enabled? && root_ancestor) || ancestors.first
```

Comment:
Is there any oddness if a user assigns to ancestor_ids?  Does that update `root_ancestor`?  Or is `ancestor_ids` not assignable?